privaterelay/management/commands/update_phone_remaining_stats.py

```diff
@@ -3,58 +3,88 @@
-            extra_data__icontains=sub_with_phone
+def get_next_reset_date(profile: Profile) -> datetime:
+    # TODO: consider moving this as a property in Profile model
+    # assumes that profile being passed have already been checked to have
+    # phone subscription or a free phone user
+    if profile.date_phone_subscription_reset is None:
+        # there is a problem with the sync_phone_related_dates_on_profile
+        # or a new foxfooder whose date_phone_subscription_reset did not get set in
+        logger.error(
+            "phone_user_profile_dates_not_set",
+            extra={
+                "fxa_uid": profile.fxa.uid,
+                "date_subscribed_phone": profile.date_phone_subscription_end,
+                "date_phone_subscription_start": profile.date_phone_subscription_start,
+                "date_phone_subscription_reset": profile.date_phone_subscription_reset,
+                "date_phone_subscription_end": profile.date_phone_subscription_end,
+            },
         )
-        accounts_with_phones.extend(list(social_accounts))
+        return datetime.now(timezone.utc) - timedelta(minutes=15)
+
+    calculated_next_reset_date = profile.date_phone_subscription_reset + timedelta(
+        settings.MAX_DAYS_IN_MONTH
+    )
+    if profile.date_phone_subscription_end is None:
+        return calculated_next_reset_date
+    if profile.date_phone_subscription_end < calculated_next_reset_date:
+        # return the past or the closest next reset date
+        return profile.date_phone_subscription_end
+    return calculated_next_reset_date
+
+
+def update_phone_remaining_stats() -> tuple[int, int]:
+    accounts_with_phones = get_phone_subscriber_social_accounts()
+    free_phones_social_accounts = get_free_phone_social_accounts()
+    accounts_with_phones.update(free_phones_social_accounts)
+
+    if not settings.PHONES_ENABLED or len(accounts_with_phones) == 0:
+        return 0, 0
+
     updated_profiles = []
+    datetime_now = datetime.now(timezone.utc)
     for social_account in accounts_with_phones:
         profile = social_account.user.profile
-        # Has it been more than 30 days since we last checked the user's phone
-        # subscription?
-        if (
-            not profile.date_phone_subscription_checked
-            or profile.date_phone_subscription_checked
-            <= datetime.now(timezone.utc) - timedelta(settings.DAYS_PER_BILLING_CYCLE)
-        ):
-            # Re-fetch all FXA data into profile so we can see if the user still has
-            # a phone subscription.
-            update_fxa(social_account)
-            profile.date_phone_subscription_checked = datetime.now(timezone.utc)
+        next_reset_date = get_next_reset_date(profile)
+        if next_reset_date <= datetime_now:
+            reset_phone_remaining_stats(profile.user)
+            profile.date_phone_subscription_reset = datetime_now
             profile.save()
             updated_profiles.append(profile)
```

Comment:
quibble (non-blocking): We could improve readability here by reducing the indentation level by inverting the logic:

```suggestion
        if not next_reset_date <= datetime_now:
            continue
        reset_phone_remaining_stats(profile.user)
        profile.date_phone_subscription_reset = datetime_now
        profile.save()
        updated_profiles.append(profile)
```

Reply:
Resolved in 7e6a1b